interchain-token-service/src/user_functions.rs

```diff
@@ -385,10 +294,19 @@ pub trait UserFunctionsModule:
                 );
                 require!(second_payment.amount == gas_amount, "Invalid gas value");
 
+                // If two ESDTs were sent, check if EGLD was sent in MultiESDT and convert to EGLD identifier
+                let gas_token = if second_payment.token_identifier.as_managed_buffer()
```

Comment:
`second_payment.is_egld()`, could be used.

Reply:
We are using an older framework version that doesn't know about EGLD in MultiESDT, hence we need to manually check this. Not upgrading the framework because the other contracts were audited with this version.